internal/benchrunner/runners/rally/runner.go

```diff
@@ -274,6 +341,43 @@ func (r *runner) setUp() error {
 	return nil
 }
 
+func (r *runner) extractSimulatedTemplate(indexTemplate string) (string, error) {
+	simulateTemplate, err := r.options.ESAPI.Indices.SimulateTemplate(r.options.ESAPI.Indices.SimulateTemplate.WithName(indexTemplate))
+	if err != nil {
+		return "", fmt.Errorf("error simulating template from composable template: %s: %w", indexTemplate, err)
+	}
+
```

Comment:
Check also the status code of the response.
```suggestion

	if  simulateTemplate.IsError() {
		return "", fmt.Errorf("error simulating template from composable template: %s: %s", indexTemplate, simulateTemplate.String())
	}
```